handlers/product-move-api/src/main/scala/com/gu/productmove/endpoint/move/switchtype/ToRecurringContribution.scala

```diff
@@ -75,11 +77,24 @@ class ToRecurringContributionImpl(
       // Make sure that price is valid and acceptable
       _ <- validateOldMembershipPrice(currency, billingPeriod, price)
 
+      // Check if we need term renewal to avoid "cancellation date cannot be later than term end date" error
+      // We need to perform term renewal if the chargedThroughDate (which will be used as the remove date)
+      // extends beyond the current subscription term end date.
+      chargedThroughDate = activeRatePlanCharge.chargedThroughDate.get
+      needsTermRenewal = activeRatePlanCharge.chargedThroughDate.exists(_.isAfter(subscription.termEndDate))
+      _ <- ZIO.when(needsTermRenewal) {
+        ZIO.log(
+          s"Performing term renewal because chargedThroughDate $chargedThroughDate is after termEndDate ${subscription.termEndDate}",
+        ) *>
+          termRenewal.renewSubscription(subscriptionName, runBilling = false) *>
+          ZIO.log(s"Term renewal completed for subscription $subscriptionName")
```

Comment:
what is `*>` for, is it a way to avoid a for comprehension and discard the result?

It does look concise and ZIO encourages it, but in the past we've shied away from ungooglable symbols where possible.

optional: Happy either way, but here is what I would have written, the main benefit is for compehension is very standard
```suggestion
        for {
          _ <- ZIO.log(
          s"Performing term renewal because chargedThroughDate $chargedThroughDate is after termEndDate ${subscription.termEndDate}",
        )
          renewal <- termRenewal.renewSubscription(subscriptionName, runBilling = false) *>
          _ <- ZIO.log(s"Term renewal completed for subscription $subscriptionName")
        } yield renewal
```